src/Agent.Plugins/PipelineCache/PipelineCacheTaskPluginBase.cs

```diff
@@ -81,26 +85,35 @@ public async Task RunAsync(AgentTaskPluginExecutionContext context, Cancellation
             string salt = saltValue?.Value ?? string.Empty;
 
             VariableValue workspaceRootValue = context.Variables.GetValueOrDefault("pipeline.workspace");
-            string worksapceRoot = workspaceRootValue?.Value;
+            string workspaceRoot = workspaceRootValue?.Value;
 
             string key = context.GetInput(PipelineCacheTaskPluginConstants.Key, required: true);
             string restoreKeysBlock = context.GetInput(PipelineCacheTaskPluginConstants.RestoreKeys, required: false);
 
-            (bool isOldFormat, string[] keySegments, IEnumerable<string[]> restoreKeys) = ParseIntoSegments(salt, key, restoreKeysBlock);
+            (bool isOldFormat, IEnumerable<string> keySegments, IEnumerable<string[]> restoreKeys) = ParseIntoSegments(salt, key, restoreKeysBlock);
 
             if (isOldFormat)
             {
                 context.Warning(OldKeyFormatMessage);
             }
 
+            VariableValue packValue = context.Variables.GetValueOrDefault(PackingVariableName);
+            string pack = packValue?.Value ?? string.Empty;
+
+            if(String.IsNullOrWhiteSpace(pack))
+            {
+                string segment = isWindows ? "\"microsoft.azure.pipelines.caching.pack=7z\"" : "\"microsoft.azure.pipelines.caching.pack=tar\"" ;
```

Comment:
I'm concerned about this because the size of the key is limited

Reply:
Yup, open to changing it. @willsmythe

Reply:
ms.tar = true is the suggestion, what's the size limitation btw?

Reply:
As we are packing just the tar, this won't be platform specific.